frontend/src/Components/HomePage/AddTrip/FlightForm.jsx

```diff
@@ -50,31 +29,31 @@ const FlightForm= () => {
     const handleSubmit = async (e) => {
 
         e.preventDefault();
-        setFlightStatus(null);
         setError(null);
         setIsLoading(true);
-        setFlightAdded(false);  
+        updateFlightData({ flightStatus: null, flightAdded: false });  
     
         try {
             const response = await axios.get(`${API_BASE_URL}/flight-status`, {
             params: {
-                carrierCode: carrierCode,
-                flightNumber: flightNumber,
-                scheduledDepartureDate: departureDate
+                carrierCode: flightData.carrierCode,
+                flightNumber: flightData.flightNumber,
+                scheduledDepartureDate: flightData.departureDate
             },
             withCredentials: true
 
             });
             
             if (response.data && response.data.data && response.data.data.length > 0) {
-                setFlightStatus(response.data);
+                updateFlightData({ flightStatus: response.data });
             } else {
                 setError("No flights found. Please check your input and try again.");
             }
         
         } catch (err) {
             if (err.response && err.response.status === 401) {
```

Comment:
I would suggest to create a helper function to handle this, so that you can reuse this check every time you catch an API exception.